app/src/main/java/org/kiwix/kiwixmobile/main/AddNoteDialog.java

```diff
@@ -0,0 +1,427 @@
+package org.kiwix.kiwixmobile.main;
+
+import android.Manifest;
+import android.app.Dialog;
+import android.content.Context;
+import android.content.DialogInterface;
+import android.content.Intent;
+import android.content.pm.PackageManager;
+import android.net.Uri;
+import android.os.Build;
+import android.os.Bundle;
+import android.os.Environment;
+import android.text.Editable;
+import android.text.TextWatcher;
+import android.util.Log;
+import android.view.LayoutInflater;
+import android.view.MenuItem;
+import android.view.View;
+import android.view.ViewGroup;
+import android.view.inputmethod.InputMethodManager;
+import android.widget.EditText;
+import android.widget.TextView;
+import android.widget.Toast;
+
+import androidx.annotation.NonNull;
+import androidx.annotation.Nullable;
+import androidx.appcompat.app.AlertDialog;
+import androidx.appcompat.widget.Toolbar;
+import androidx.core.content.ContextCompat;
+import androidx.core.content.FileProvider;
+import androidx.fragment.app.DialogFragment;
+
+import org.kiwix.kiwixmobile.BuildConfig;
+import org.kiwix.kiwixmobile.KiwixApplication;
+import org.kiwix.kiwixmobile.R;
+import org.kiwix.kiwixmobile.data.ZimContentProvider;
+import org.kiwix.kiwixmobile.utils.SharedPreferenceUtil;
+
+import java.io.BufferedReader;
+import java.io.File;
+import java.io.FileOutputStream;
+import java.io.IOException;
+
+/**
+ * Created by @Aditya-Sood (21/05/19) as a part of GSoC 2019
+ *
+ * AddNoteDialog extends DialogFragment and is used to display the note corresponding to a particular
+ * article (of a particular zim file/wiki/book) as a full-screen dialog fragment.
+ *
+ * Notes are saved as text files at location: "{External Storage}/Kiwix/Notes/ZimFileTitle/ArticleTitle.txt"
+ * */
+
+public class AddNoteDialog extends DialogFragment {
+
+  private SharedPreferenceUtil sharedPreferenceUtil = new SharedPreferenceUtil(KiwixApplication.getInstance());
+
+  private Toolbar toolbar;  // Displays options for the note dialog
+  private TextView addNoteTextView; // Displays article title
+  private EditText addNoteEditText; // Displays zim file title (wiki name)
+  private String zimFileTitle;
+  private String articleTitle;
+  private boolean noteFileExists = false;
+  private boolean noteEdited = false; // Keeps track of state of the note (whether edited since last save)
+
+  private final String TAG = "AddNoteDialog";
+
+  @Override
+  public void onCreate(Bundle savedInstanceState) {
+    super.onCreate(savedInstanceState);
+
+    if(sharedPreferenceUtil != null && sharedPreferenceUtil.nightMode()) {
```

Comment:
sharedPreferenceUtil can't be null so I would remove this check, also consider that this chunk of code is repeating itself when really it just wants to swap out a parameter. I would write:
`setStyle(DialogFragment.STYLE_NORMAL, sharedPreferenceUtil.nightMode() ? R.style.AddNoteDialogStyle_Night : R.style.AddNoteDialogStyle)`

Reply:
Refactored [here](https://github.com/kiwix/kiwix-android/pull/1198/commits/873eb2bce86c76b11962367701cb66bdf66b9f5e#diff-36e13fa4a9132616aee81e1f88bfa0ffR73)